### DISCUSSION TITLE
Can I provide an existing database connection to quartz?

### DISCUSSION THREAD

**Discussion by DevonBT** [quartznet/quartznet#1684] [Q&A]
My production environment uses Amazon RDS to authenticate to databases, so the application instances never get a password to use in DB connection strings. Instead, authentication occurs like this:

    services.AddDbContext<ServiceDbContext>((sp, dbContextBuilder) =>
    {
        var dbUserName = _configuration["RDS:DbUser"];
        var dbName = _configuration["RDS:DbName"];
        var dbserver = _configuration["RDS:DbUrl"];
        dbContextBuilder.UseNpgsql(
            $"Server={dbserver};User Id={dbUserName};Database={dbName};",
            optionsBuilder =>
            {
                optionsBuilder.ProvidePasswordCallback((host, port, _, username) => RDSAuthTokenGenerator.GenerateAuthToken(host, port, username));
            });
    });

Quartz seems to expect a connection string to open its own connection to the database, which I fear won't work with this pattern. We cannot provide a password for that connection string. 

Are there any mechanisms that will let me get a database connection into Quartz? My domain code CAN form a connection using the code above, so I think it will be possible for me to 'pluck' that connection out during ASP.NET's startup. Or something similar to the `ProvidePasswordCallback` used in configuring Npgsql? I'm not an expert on RDS, so I wonder if it's possible that a password-less connection attempt from Quartz will succeed after the domain code authenticates properly and somehow 'authenticates' the entire container?

Thanks for any help with this, I'm not sure what approach should be used but I'm open to any suggestions.

**Comment by lahma** [comment]
There was [a recent discussion where also custom IDbProvider was mentioned ](https://github.com/quartznet/quartznet/issues/1616#issuecomment-1140477206)which allows you to create the connection the way you want.
